src/app/widget/widget.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input, OnDestroy } from '@angular/core';
+import { IWidget } from '../weather.model';
+import { WidgetApiService } from '../widget-api.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-widget',
+  standalone: true,
+  imports: [],
+  templateUrl: './widget.component.html',
+  styleUrl: './widget.component.css'
+})
+export class WidgetComponent implements OnDestroy {
+  @Input() weatherData!: IWidget;
+  weatherSubscription: Subscription | undefined;
+
+  constructor(private widgetApiService: WidgetApiService) { }
+
+  refreshWeather(): void {
+    this.widgetApiService.getWeatherForWidget(this.weatherData.cityName)
```

Comment:
You not unsubscribe. You need to do in this way: 
`
this.weatherSubscription = this.widgetApiService.getWeatherForWidget.......
`